fiftyone/factory/repos/delegated_operation.py

```diff
@@ -390,6 +385,25 @@ def update_run_state(
             return_document=pymongo.ReturnDocument.AFTER,
         )
 
+        if (
+            doc
+            and doc.get("num_partitions")
+            and run_state is ExecutionRunState.FAILED
+        ):
+            # If a parent operation is failed, also mark the children as failed
+            self._collection.update_many(
+                {
+                    "group_id": doc["_id"],
+                    "run_state": {"$nin": ["failed", "completed"]},
+                },
+                {
+                    "$set": {
+                        **update["$set"],
+                        "result": {"error": "parent operation failed"},
```

Comment:
Hm I didn't think this would work since update already has "result" key defined, but it does in this syntax!
If it were `dict(**update["$set"], result={"error": "parent operation failed"})`, then it fails with `TypeError: dict() got multiple values for keyword argument 'result'`